### PR TITLE
Adjust DefiniteAssignmentPass.WriteConsideredUse for UTF8 String Literals.

### LINKED ISSUE #60319: Missing "variable is assigned but its value is never used" warning for assignment utilizing UTF8 String Literal
```
class Program
{
    static void Main()
    {
        var x = "123"U8; // <- Missing a warning here?
        var y = "abc"; // warning CS0219: The variable 'y' is assigned but its value is never used

        byte[] z1 = "345"; // <- Missing a warning here?
        System.Span<byte> z2 = "678"; // warning CS0219: The variable 'z2' is assigned but its value is never used
        System.ReadOnlySpan<byte> z3 = "678"; // warning CS0219: The variable 'z3' is assigned but its value is never used
    }
}
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #60319.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs [hunk 1]
```diff
@@ -799,7 +799,10 @@ protected virtual void NoteWrite(Symbol variable, BoundExpression value, bool re
         internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
         {
             if (value == null || value.HasAnyErrors) return true;
-            if ((object)type != null && type.IsReferenceType && type.SpecialType != SpecialType.System_String)
+
+            if ((object)type != null && type.IsReferenceType &&
+                type.SpecialType != SpecialType.System_String &&
+                (type is not ArrayTypeSymbol { IsSZArray: true } arrayType || arrayType.ElementType.SpecialType != SpecialType.System_Byte))
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs:805]: Consider combining into a single pattern like `type is not ArrayTypeSymbol { IsSZArray: true, ElementType.SpecialType: SpecialType.System_Byte }`

File: src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs [hunk 2]
```diff
@@ -3384,6 +3384,77 @@ static void Main()
...
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        const string stringNull = null;
+
+        var x1 = ""123""U8;
+        Span<byte> x2 = ""124""U8;
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs:3399]: Is it intentional that the warning is not given here even though it is given on 'z2'?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs:3399]: > Is it intentional that the warning is not given here even though it is given on 'z2'?

This is similar to ```System.ReadOnlySpan<char> x3 = "abc";```. A user defined conversion considered as a usage.
 On `z2` there is no user-defined conversion from the language point of view, there is a built-in conversion.

**Review by RikkiGibson — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review.

**Pull request merged**